src/playground/service.clj

```diff
@@ -33,7 +33,10 @@
   (ring-resp/response (views/home request)))
 
 (defn insert-page [request]
-  (ring-resp/response (views/insert request)))
+  (if (authenticated? (:session request))
```

Comment:
Perhaps if in the views `header-links` https://github.com/manandearth/playground/blob/061049588221a3dc6589bd5e3e8bbf6b09ad3ebf/src/playground/views.clj#L12
I move the href of `:invoices-insert` to the authenticated :div, This authentication in the handler will be redundant.
Is that better design?

Reply:
Not sure. Imagine I'm a hacker and want to use the handler. The lack of a href won't stop me; authorization in the handler will.